sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -320,6 +319,65 @@ def test_sample_weight_missing():
     assert_raises(ValueError, clf.fit, X, y_regr)
 
 
+def test_sample_weight_adaboost_regressor():
+    """
+    AdaBoostRegressor should work without sample_weights in the base estimator
+
+    The random weighted sampling is done internally in the _boost method in
+    AdaBoostRegressor.
+    """
+    class DummyEstimator(BaseEstimator):
+
+        def fit(self, X, y):
+            pass
+
+        def predict(self, X):
+            return np.zeros(len(X))
+
+    boost = AdaBoostRegressor(DummyEstimator(), n_estimators=3)
+    boost.fit(X, y_regr)
+    assert_equal(len(boost.estimator_weights_), len(boost.estimator_errors_))
+
+
+def test_multidimensional_X():
+    class DummyClassifier(BaseEstimator):
+
+        def fit(self, X, y, sample_weight=None):
+            self.classes_ = np.unique(y)
+            return self
+
+        def predict(self, X):
+            n_samples = len(X)
+            predictions = np.random.choice(self.classes_, n_samples)
+            return predictions
+
+        def predict_proba(self, X):
+            n_samples = len(X)
+            n_classes = len(self.classes_)
+            probas = np.random.randn(n_samples, n_classes)
+            return probas
+
+    class DummyRegressor(BaseEstimator):
+
+        def fit(self, X, y):
+            return self
+
+        def predict(self, X):
+            n_samples = len(X)
+            predictions = np.random.randn(n_samples)
+            return predictions
+
+    X = np.random.randn(50, 3, 3).tolist()
```

Comment:
At first I thought that we should remove the `check_X_y` and `check_array` so I thought to just pass an array. I removed the conversion since I retain the checks.